### PR TITLE
Fix: Blocks of multi-line inline elements are too large

### PR DESCRIPTION
**Pull request by igrep**
Problem
====

Blocks of multi-line inline elements can be broken even if they don't look collided with the ball in some cases.

Here's an example reproduced by the `controlMode: "mouse"` feature created in https://github.com/canalun/brick-break-anywhere/pull/15, and hard-coding `visualizeBlocks: false` (checkout [my debug branch](https://github.com/igrep/brick-break-anywhere/tree/debug) if you reproduce by yourself):

bba-too-large-block

As you can see, the larger block `12345678901234567890` got broken just as the `12345` got hit.

<details>
  <summary>The source of the screenshot HTML is here.</summary>

```html
<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="Content-Language" content="ja">
  </head>
  <body style="width: 8em; font-size: 50pt; font-family: monospace; line-break: anywhere">
    <span>12345678901234567890</span>
    <span>12345</span>
  </body>
</html>
```
</details>

Cause
----

This is because the blocks of elements are created from the rectangles returned by [`getBoundingClientRect`](https://developer.mozilla.org/ja/docs/Web/API/Element/getBoundingClientRect), which returns only one rectangle from an element. `getBoundingClientRect` doesn't have a problem with most elements, each of which has only one bounding rectangle. But given a multi-line element, `getBoundingClientRect` returns an unexpectedly large rectangle including its rightmost bottom edge.

Solution
====

Use [`getClientRects`](https://developer.mozilla.org/ja/docs/Web/API/Element/getClientRects) instead of `getBoundingClientRect` to create blocks. It returns all rectangles of elements including multi-line elements.

NOTE
====

- I recommend you to review with the ignore-spaces mode, and you'll see the changes in updateBall.ts are safe.
- I tested this change only in the example case and [my website's top page](https://the.igreque.info/). Tell me if you think of any side-effect.
- 👍Thanks for reading to the bottom line! This is the last pull request so far. I don't mean I'll stop contributing after this, but I won't do anymore for the time being.  🙇お忙しい中ありがとうございます！お疲れ様です🍵

### CODE REVIEW

File: src/game/object/blocks.ts [hunk 1]
```diff
@@ -27,42 +29,34 @@ function convertElementToBlock(element: Element): Block {
...
-  rect: {
-    top: number
-    bottom: number
-    left: number
-    right: number
-  }
-) {
+  rect: Rect
+): Rect {
+  let result = { ...rect };
```
Comment (canalun) [inline src/game/object/blocks.ts]:
```suggestion
  const result = { ...rect };
```
Properties of 'const' objects can be changed, so I think you can use 'const' here :)

Comment (canalun) [reply src/game/object/blocks.ts]: If you wanna use 'let' as a marker for the usage of while loop, is the below also an idea?
```suggestion
  let { top, bottom, left, right } = rect
```

**Review by canalun — COMMENTED** [review]
Thank you so much for finding this bug! It's really tricky and I overlooked it.
Many thanks!!!

> 👍Thanks for reading to the bottom line! This is the last pull request so far. I don't mean I'll stop contributing after this, but I won't do anymore for the time being. 🙇お忙しい中ありがとうございます！お疲れ様です🍵

I am so grateful that words can't express my appreciation...! This app improves a lot thanks to you, igrep-san.

I look forward to enjoying happy-coding with you somewhere again!!
Of course, you're always welcome to come back to this repo whenever you want :)

File: src/game/object/blocks.ts (hunk 1 above)
Comment (igrep) [reply src/game/object/blocks.ts]: Nice catch! I'll adopt the former.

Comment (canalun) [reply src/game/object/blocks.ts]: thanks!

**Pull request merged**